src/app/routes/enrolment/view-requests/issuer-requests/issuer-requests.component.ts

```diff
@@ -83,8 +83,8 @@ export class IssuerRequestsComponent
     this.dialogRef.close(true);
   }
 
-  updateExpirationDate(validityPeriod: number): void {
-    this.expirationTime = Date.now() + validityPeriod;
+  updateExpirationDate(validityPeriod: number | undefined): void {
+    this.expirationTime = validityPeriod ? Date.now() + validityPeriod : null;
```

Comment:
IMO, this increases the complexity of the function a bit too much and it is better to avoid the use of `undefined`/`null` as a parameter if possible as it is hard to guess how the function will behave by reading its signature. In order words, by just reading `updateExpirationDate(validityPeriod: number | undefined): void`, it's hard to guess what the behaviour will be if `undefined` is passed. Does that mean that `expirationTime` will be set to `undefined` or does it mean that it will be set to some value and just `validityPeriod` won't be used?

I propose that there be two functions:
- `setExpirationTimeUsingValidity: void`
- `clearExpirationDate(): void`

In my opinion, this would make the code clearer.

Reply:
Implemented this